include/dlaf/eigensolver/tridiag_solver/rot.h

```diff
@@ -205,9 +205,10 @@ void applyGivensRotationsToMatrixColumns(const SizeType i_begin, const SizeType
 /// @pre mat is distributed along rows the same way as comm_row
 /// @pre memory layout of @p mat is column major.
 template <class T, Device D, class GRSender>
-void applyGivensRotationsToMatrixColumns(comm::Communicator comm_row, const comm::IndexT_MPI tag,
-                                         const SizeType i_begin, const SizeType i_end,
-                                         GRSender&& rots_fut, Matrix<T, D>& mat) {
+void applyGivensRotationsToMatrixColumns([[maybe_unused]] comm::CommunicatorGrid& grid,
+                                         common::Pipeline<comm::Communicator>& comm_row_chain,
```

Comment:
This comment just for highlighting it.

In the past we already pointed out this minor "problem" of not being able to access metadata of things "enclosed" in a sender (at the time it was a future, but it does not change much).

I don't think it is worth to over-complicate things, and I'm ok with this alternative.

Reply:
Closing in favour of the discussion starting at https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/5469465492. We will change this to not pass the full `CommunicatorGrid`.